### PR TITLE
nvdimm_hotplug: add new case to test hotplug nvdimm device

### PR DESCRIPTION
**Pull request by huangyum**
ID: 1845366
Signed-off-by: Yumei Huang <yuhuang@redhat.com>

### CODE REVIEW

**Comment by huangyum** [comment]
(1/1) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.nvdimm.nvdimm_hotplug: PASS (142.82 s)

**Comment by huangyum** [comment]
@PaulYuuu @yanan-fu Please help review, thanks.

File: qemu/tests/nvdimm.py [hunk 1]
```diff
@@ -139,6 +141,12 @@ def run(test, params, env):
         error_context.context("Login to the guest", logging.info)
         login_timeout = int(params.get("login_timeout", 360))
         nvdimm_test.session = vm.wait_for_login(timeout=login_timeout)
+        target_mems = params.get("target_mems")
+        if target_mems:
+            hotplug_test = MemoryHotplugTest(test, params, env)
+            for mem in target_mems.split():
+                hotplug_test.hotplug_memory(vm, mem)
+            time.sleep(10)
```
[the remark this answers is not in the document]
Comment (huangyum) [reply qemu/tests/nvdimm.py:149]: Yes, you are right. May take several seconds for system to process the hotplugged device.

File: qemu/tests/nvdimm.py [hunk 2]
```diff
@@ -139,6 +141,12 @@ def run(test, params, env):
         error_context.context("Login to the guest", logging.info)
         login_timeout = int(params.get("login_timeout", 360))
         nvdimm_test.session = vm.wait_for_login(timeout=login_timeout)
+        target_mems = params.get("target_mems")
+        if target_mems:
+            hotplug_test = MemoryHotplugTest(test, params, env)
+            for mem in target_mems.split():
```
[the remark this answers is not in the document]
Comment (huangyum) [reply qemu/tests/nvdimm.py]: Yeah, that works too. Updated.

**Comment by huangyum** [comment]
(1/7) Host_RHEL.m8.u3.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.nvdimm.nvdimm_basic: PASS (132.38 s)
 (2/7) Host_RHEL.m8.u3.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.nvdimm.nvdimm_emulate: PASS (115.25 s)
 (3/7) Host_RHEL.m8.u3.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.nvdimm.nvdimm_emulate.pmem_on: PASS (114.26 s)
 (4/7) Host_RHEL.m8.u3.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.nvdimm.nvdimm_dax: PASS (114.23 s)
 (5/7) Host_RHEL.m8.u3.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.nvdimm.nvdimm_mapsync: PASS (87.45 s)
 (6/7) Host_RHEL.m8.u3.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.nvdimm.nvdimm_mode: PASS (168.81 s)
 (7/7) Host_RHEL.m8.u3.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.nvdimm.nvdimm_hotplug: PASS (155.23 s)

File: qemu/tests/cfg/nvdimm.cfg [hunk 3]
```diff
@@ -26,7 +26,7 @@
     post_command = "rm -rf ${nv_backend}"
     mem_devs = mem1
     size_mem = 1G
-    share_mem = on
+    share_mem = yes
```
Comment (yanan-fu) [inline qemu/tests/cfg/nvdimm.cfg:29]: To make it 'True/False' for QEMU 5.0 ?  Have you try it with RHEL7 ?

File: qemu/tests/cfg/nvdimm.cfg [hunk 4]
```diff
@@ -83,3 +83,6 @@
             del dev_path
             del format_command
             del mount_command
+        - nvdimm_hotplug:
+            del mem_devs
```
Comment (yanan-fu) [inline qemu/tests/cfg/nvdimm.cfg:87]: With this change, the `verify_nvdimm` is meaningless:
https://github.com/autotest/tp-qemu/blob/185b121e24d3f798e81d12e26cf1992822c1613f/qemu/tests/nvdimm.py#L46

File: qemu/tests/cfg/nvdimm.cfg (hunk 3 above)
Comment (huangyum) [reply qemu/tests/cfg/nvdimm.cfg:29]: Yes, it works well on RHEL7 as well.

File: qemu/tests/cfg/nvdimm.cfg (hunk 4 above)
Comment (huangyum) [reply qemu/tests/cfg/nvdimm.cfg:87]: You are right. Updated this function, please help review again.

**Comment by huangyum** [comment]
Host_RHEL.m7.u9.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.9.x86_64.io-github-autotest-qemu.nvdimm.nvdimm_hotplug: PASS (158.37 s)

File: qemu/tests/nvdimm.py [hunk 5]
```diff
@@ -37,13 +39,14 @@ def run_guest_cmd(self, cmd, check_status=True):
...
-    def verify_nvdimm(self, vm):
+    def verify_nvdimm(self, vm, mems):
         """
         verify nvdimm in monitor and guest
 
         :params vm: VM object
+        :params mems: memory objects
         """
-        dimms_expect = set(["dimm-%s" % dev for dev in self.params.objects("mem_devs")])
+        dimms_expect = set("dimm-%s" % mem for mem in mems)
```
Comment (yanan-fu) [inline qemu/tests/nvdimm.py:49]: Suggest to print the dimms_expect, to avoid the issue before: test pass, debug log is normal but miss one check point.

Comment (huangyum) [reply qemu/tests/nvdimm.py:49]: Done, thanks.

**Comment by yanan-fu** [comment]
Acked-by: Yanan Fu <yfu@redhat.com>

@PaulYuuu  Could you help review again, thanks!

**Comment by PaulYuuu** [comment]
Ack

**Pull request merged**